[#1802, #1803, #1804, #1876] Added HTML App Menu for Windows

tonyanziano opened the pull request:
#1876, #1802, #1803, #1804

===

![image](https://user-images.githubusercontent.com/3452012/66233301-ddfc5180-e69f-11e9-844f-334ff6086f85.png)

![image](https://user-images.githubusercontent.com/3452012/66233331-f3717b80-e69f-11e9-8e4f-e160018916fd.png)

![image](https://user-images.githubusercontent.com/3452012/66233358-ff5d3d80-e69f-11e9-9090-fd0d3953b2ed.png)

corinagum commented on packages/sdk/ui-react/src/widget/menu/menuButton.tsx:
```diff
@@ -0,0 +1,163 @@
+  private buttonRef: HTMLButtonElement;
+  private menuButtonId: string;
+  private onBodyClickListener: () => void;
+  private onMenuItemSelectedListener: () => void;
```
lines 57 and 58 alphabetize?

tonyanziano commented on packages/sdk/ui-react/src/widget/menu/menu.scss:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    > .menu-item-check {
+      background: var(--menu-item-hover-color);
```
good catch!

tonyanziano commented on packages/app/client/src/ui/shell/appMenu/appMenuTemplate.ts:
```diff
@@ -0,0 +1,260 @@
+            }
+          });
+        },
+        subtext: `${CtrlOrCmd}+-`,
```
Honestly, I'm not entirely sure. Like you pointed out, the keybind already existed like that so I just carried it over.

corinagum replied:
Looking at VS Code, both `Ctrl/Cmd =` and `Ctrl/Cmd Shift =` are zoom in, and likewise Ctrl/Ctrl Shift for - are zoom out. Should Emulator be updated to the same?

tonyanziano replied:
Added shortcuts for both. 👍

tonyanziano commented:
@corinagum Addressed your comments.

Thanks for the note on the menu accessibility, I didn't even know you could use `Ctrl+<arrow>` to navigate lists, so I didn't test any of that.

My thought is that we submit it as is, and then if the a11y team ends up dinging us for that, we can fix it. However, to my memory, I don't think they reported any issues involving `Ctrl+<arrow>` navigation, but maybe we just didn't have any instances like the app menu.
